src/runtime/classbase.cs

```diff
@@ -246,6 +246,44 @@ public static IntPtr tp_str(IntPtr ob)
             }
         }
 
+        public static IntPtr tp_repr(IntPtr ob)
+        {
+            var co = GetManagedObject(ob) as CLRObject;
+            if (co == null)
+            {
+                return Exceptions.RaiseTypeError("invalid object");
+            }
+            try
+            {
+                //if __repr__ is defined, use it
+                var instType = co.inst.GetType();
+                System.Reflection.MethodInfo methodInfo = instType.GetMethod("__repr__");
+                if (methodInfo != null && methodInfo.IsPublic)
+                {
+                    var reprString = methodInfo.Invoke(co.inst, null) as string;
+                    return Runtime.PyString_FromString(reprString);
+                }
+
+                //otherwise use the standard object.__repr__(inst)
+                IntPtr args = Runtime.PyTuple_New(1);
+                Runtime.PyTuple_SetItem(args, 0, ob);
+                IntPtr reprFunc = Runtime.PyObject_GetAttrString(Runtime.PyBaseObjectType, "__repr__");
+                var output =  Runtime.PyObject_Call(reprFunc, args, IntPtr.Zero);
+                Runtime.XDecref(args);
```

Comment:
Decreasing args reference count actually decreases ob reference count, which marks it as garbage for the GC!
After a while GC frees the object which causes the program to crash if the object is used again.
One fix for this can be adding the line:
`Runtime.XIncref(ob);`
at line 269 (before setting ob to the tuple's first argument).

Reply:
Thanks a lot for catching this. Would you prepare a small PR?

Reply:
Done
[1160](https://github.com/pythonnet/pythonnet/pull/1160)